[inductor] Make clone_graph copy node name as well

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103557
* __->__ #103409

Summary: This solves an inconsistency between two-pass fusion results
when turning on cpp wrapper. The unit test comes from yolov3.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @aakhundov @chenyang78

desertfire committed:
[inductor] Experiment

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Make clone_graph copy node name as well"


Summary: This solves the inconsistency bewteen two-pass fusion results
when turning on cpp wrapper. The unit test comes from yolov3.

[ghstack-poisoned]

eellison commented on torch/_inductor/pattern_matcher.py:
```diff
@@ -1046,6 +1046,7 @@ def run_node(self, old_node):
             new_node = super().run_node(old_node)
             if isinstance(new_node, torch.fx.Proxy):
                 new_node.node.meta.update(old_node.meta)
+                new_node.node.name = old_node.name
```
I think you should use this API instead to safe  https://github.com/pytorch/pytorch/blob/main/torch/fx/graph.py#L840

eellison reviewed (commented):
i think we want to use the existing name api in case there is a conflict

eellison reviewed (approved):
looks good with api change

desertfire committed:
Update on "[inductor] Make clone_graph copy node name as well"


Summary: This solves an inconsistency between two-pass fusion results
when turning on cpp wrapper. The unit test comes from yolov3.

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge

desertfire committed:
Update on "[inductor] Make clone_graph copy node name as well"


Summary: This solves an inconsistency between two-pass fusion results
when turning on cpp wrapper. The unit test comes from yolov3.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 aakhundov

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge

desertfire commented:
@pytorchbot rebase -b main

desertfire committed:
Update on "[inductor] Make clone_graph copy node name as well"


Summary: This solves an inconsistency between two-pass fusion results
when turning on cpp wrapper. The unit test comes from yolov3.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 aakhundov chenyang78

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge

desertfire committed:
Update on "[inductor] Make clone_graph copy node name as well"


Summary: This solves an inconsistency between two-pass fusion results
when turning on cpp wrapper. The unit test comes from yolov3.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 aakhundov chenyang78

[ghstack-poisoned]

osalpekar commented:
@pytorchbot revert -m "torchbench regression starting this commit. See https://www.torch-ci.com/pytorch/pytorch/commit/2d745b95d723641e575027bd4e2fff612f61cc8f for more info" -c landrace
